src/main/java/liquibase/ext/databricks/snapshot/jvm/TableSnapshotGeneratorDatabricks.java

```diff
@@ -0,0 +1,60 @@
+package liquibase.ext.databricks.snapshot.jvm;
+
+import liquibase.Scope;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.executor.ExecutorService;
+import liquibase.ext.databricks.database.DatabricksDatabase;
+import liquibase.snapshot.DatabaseSnapshot;
+import liquibase.snapshot.jvm.TableSnapshotGenerator;
+import liquibase.statement.core.RawParameterizedSqlStatement;
+import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.Table;
+
+import java.util.List;
+import java.util.Map;
+
+public class TableSnapshotGeneratorDatabricks extends TableSnapshotGenerator {
+
+    private final static String LOCATION = "Location";
+    private final static String TABLE_PROPERTIES = "Table Properties";
+    private final static String TBL_PROPERTIES = "tblProperties";
+    private final static String DETAILED_TABLE_INFORMATION_NODE = "# Detailed Table Information";
+
+    @Override
+    public int getPriority(Class<? extends DatabaseObject> objectType, Database database) {
+        if (database instanceof DatabricksDatabase)
+            return PRIORITY_DATABASE;
+        return PRIORITY_NONE;
+
+    }
+
+    @Override
+    protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot snapshot) throws DatabaseException {
+        Table table = (Table) super.snapshotObject(example, snapshot);
+        Database database = snapshot.getDatabase();
+
+        String query = String.format("DESCRIBE TABLE EXTENDED %s.%s.%s;", database.getDefaultCatalogName(), database.getDefaultSchemaName(), example.getName());
+        List<Map<String, ?>> tablePropertiesResponse = Scope.getCurrentScope().getSingleton(ExecutorService.class)
+                .getExecutor("jdbc", database).queryForList(new RawParameterizedSqlStatement(query));
+        // DESCRIBE TABLE EXTENDED returns both columns and additional information.
+        // We need to make sure "Location" is not column in the table, but table location in s3
+        boolean detailedInformationNode = false;
+        for (Map<String, ?> tableProperty : tablePropertiesResponse) {
+            if (tableProperty.get("COL_NAME").equals(DETAILED_TABLE_INFORMATION_NODE)) {
+                detailedInformationNode = true;
+                continue;
+            }
+            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(LOCATION)) {
+                table.setAttribute(LOCATION, tableProperty.get("DATA_TYPE"));
+            }
+            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(TABLE_PROPERTIES)) {
+                String tblProperties = (String) tableProperty.get("DATA_TYPE");
+                table.setAttribute(TBL_PROPERTIES, tblProperties.substring(1, tblProperties.length() - 1));// remove starting and ending square brackets
```

Comment:
Would make sense to add a `break` here to leave the for loop as we already found what we want?

Reply:
I'm unsure if we can rely on the order of `Location` and `tblProperties` rows to break in second block. I can create a var that will be incremented each time we find needed object and break if all are found
Something like
```
        byte nodesFound = 0;
        for (Map<String, ?> tableProperty : tablePropertiesResponse) {
            if (tableProperty.get("COL_NAME").equals(DETAILED_TABLE_INFORMATION_NODE)) {
                detailedInformationNode = true;
                continue;
            }
            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(LOCATION)) {
                table.setAttribute(LOCATION, tableProperty.get("DATA_TYPE"));
                nodesFound++;
            }
            if (detailedInformationNode && tableProperty.get("COL_NAME").equals(TABLE_PROPERTIES)) {
                String tblProperties = (String) tableProperty.get("DATA_TYPE");
                table.setAttribute(TBL_PROPERTIES, tblProperties.substring(1, tblProperties.length() - 1));// remove starting and ending square brackets
                nodesFound++;
            }
            if (nodesFound == 2) {
                break;
            }
        }
        ```

Reply:
Oh I see. So no worries, makes sense now!